tests/testsuite.src/run_file.at

```diff
@@ -13029,3 +13029,90 @@ Z9
 ])
 
 AT_CLEANUP
+
+
+
+
+# Verify that FD GLOBAL will create default handlers in sub-programs, but
+# only in its own sub-programs
+
+AT_SETUP([Scope of FD GLOBAL in nested programs])
+AT_KEYWORDS([file])
+AT_DATA([prog.cob],[
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. NESTED-FILES.
+       PROCEDURE DIVISION.
+         CALL    "NESTED-PROGRAM-1".
+         CALL    "NESTED-PROGRAM-2".
+         STOP RUN.
+
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. NESTED-PROGRAM-1.
+       ENVIRONMENT DIVISION.
+       INPUT-OUTPUT SECTION.
+       FILE-CONTROL.
+         SELECT FILE-EXT ASSIGN "./TEST-FILE".
+       DATA DIVISION.
+       FILE SECTION.
+       FD FILE-EXT GLOBAL
+                 RECORD IS VARYING IN SIZE  CHARACTERS.
+       01 FILE-EXT-REC PIC X(5).
+       WORKING-STORAGE SECTION.
+       01 ERRORS             USAGE BINARY-LONG VALUE 0.
+       PROCEDURE DIVISION.
+       DECLARATIVES.
+       P01 SECTION.
+          USE AFTER ERROR PROCEDURE ON FILE-EXT.
+       P0101.
+          ADD 1 TO ERRORS.
+       P02 SECTION.
+          USE AFTER ERROR PROCEDURE ON OUTPUT.
+       P0201.
+          ADD 1 TO ERRORS.
+       END DECLARATIVES.
+       MAIN SECTION.
+         OPEN OUTPUT FILE-EXT.
+         CALL    "NESTED-PROGRAM-1-1".
+         CALL    "NESTED-PROGRAM-1-2".
+         CLOSE FILE-EXT.
+         DISPLAY "ERRORS = " ERRORS.
+         GOBACK.
+
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. NESTED-PROGRAM-1-1.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       PROCEDURE DIVISION.
+         NESTED-PROGRAM .
+         WRITE FILE-EXT-REC FROM "Hello".
+         GOBACK.
+       END PROGRAM NESTED-PROGRAM-1-1.
+
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. NESTED-PROGRAM-1-2.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01 FILE-EXT-DAT PIC X(347).
```

Comment:
Seems unused, so maybe drop the complete `DATA DIVISION` in `NESTED-PROGRAM-1-2`, `NESTED-PROGRAM-1-1` and `NESTED-PROGRAM-2`?
Do you want to do anything with ``NESTED-PROGRAM-2`? If not then please drop it.

Reply:
I think it's better to keep this second program, so that the test may detect more errors in the future.